public/content/documentation/web/page-level/form.md

```diff
@@ -1,80 +1,603 @@
+     - I HEAR It is discoverable with screenreader shortcuts as a form landmark along with its name
+     - I HEAR the screenreader switch from browse shortcuts to forms shortcuts
 
 Full information: [https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/page-level/form](https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/page-level/form)
 
 ## Developer Notes
 
-### Name
-
-- Typically doesn’t have a name or description since there must be only one instance per page.
+### Do not auto focus inputs
+
+Automatically moving focus to an input field is very confusing for people using assistive technology.
+
+### Group inputs
+
+Use `fieldset` and `legend` to group related fields, such as:
+- Sign in
+- Shipping address
+- Payment information
+
+### Error handling
+
+- Individual inputs must have programmatically described errors read by the screen reader on focus.
+- For long forms, list all errors in an alert with links back to each invalid input on submission attempts.
+
+### Field width affordance
+
+- Inputs should visually reflect the expected data length.
+    - Middle initial: 1 character
+    - State abbreviation: 2 characters
+    - Zip code: 5 characters
+    - PIN: depends on expected digits
+
+### Layout: Stack inputs vertically
+
+Avoid multiple columns:
+- Vertical scrolling is more accessible and expected.
+- Screen magnifiers make it difficult to find right-column fields.
+- Submit buttons should not be placed in sidebars only.
+
+### Use autocomplete
+
+- Enables quicker completion for all users.
+- Especially useful for people with motor disabilities.
+
+## Code examples
+
+### Use semantic HTML
+
+```html
+<form aria-label="Sign in">
+  <fieldset>
+    <legend>Sign in</legend>
+    <label for="username">Username</label>
+    <input type="text" id="username">
+    <button type="submit">Sign in</button>
+  </fieldset>
+</form>
+```
+
+<example>
+    <form aria-label="Sign in">
+      <fieldset>
+        <legend>Sign in</legend>
+        <label for="username">Username</label>
+        <input type="text" id="username">
+        <button class="Magentaa11y-button Magentaa11y-button--primary" type="submit">Sign in</button>
+      </fieldset>
+    </form>
+</example>
+
+
+### Credit card information
+
+Use `fieldset` and `legend` to group related fields, such as:
+- This form uses minimal unobtrusive masking to make the credit card number more readable. (When done poorly, masking can can cause the field to be read repeatedly as the mask refreshes)
+- Autofill attributes to help customers complete fields with less effort.
+- Using inputmode="numeric" brings up the numeric keyboard on mobile devices making entry easier.
+
+```html
+<form aria-label="Payment information">
+  <fieldset>
+    <legend>
+      Enter your payment information
+    </legend>
+    <div class="card-number-container">
+      <label for="card-number">Card Number</label>
+      <button type="button" class="cc-camera secondary">
```

Comment:
Let's add a space on either side of this <button>.

Reply:
unfortunately this breaks the markdown/file so we cannot do this